src/remote.h

```diff
@@ -36,6 +36,15 @@ struct git_remote {
 	int passed_refspecs;
 };
 
+typedef struct git_remote_connection {
+	const git_strarray *custom_headers;
+	const git_proxy_options *proxy;
+} git_remote_connection;
```

Comment:
Sorry to be nitpicky here, but I've just realized that this structure is not a connection, but instead rather an options structure. So `git_remote_connection_opts` might be more fitting? Especially as you also embed this struct below the `/* options */` comment in the `git_push` structure.